bitcoin/psbt.h

```diff
@@ -32,6 +32,8 @@ struct wally_psbt *new_psbt(const tal_t *ctx,
  */
 bool psbt_is_finalized(struct wally_psbt *psbt);
 
+struct wally_tx *psbt_finalize(const struct wally_psbt *psbt, bool finalize_in_place);
```

Comment:
Why is the `psbt` argument declared `const` if we can modify it at all in-place? Shouldn't we rather communicate the non-constness to the callers, rather than working around it being a const here?

Could also be that my mental model of `const` is wrong :-)

Reply:
Oh good point. I guess I meant it more as a "hey you better really be done with this PSBT" marker, but as you point out, since the `finalize in place` option actually updates it internally that's not really *enforceable*.

Reply:
I'll remove it, I don't think the usage corresponds to the intention of `const`